### ISSUE TITLE
Add support to alert on the result of an outer join expression

### ISSUE THREAD

**Issue by kitos9112** [grafana/grafana#92933]
**Why is this needed**:

Given two datasource queries, A and B, within a single alert rule that are later reduced in another couple of queries, C and D.
I would like to compare the labelled results of $C and $D and finally, alert if and only if there are series that match the outer join between those two.

**What would you like to be added**:

As per the title states. I am perfectly capable of doing binary and unary operations between the results of two reduced queries using expressions, but I cannot simply alert when there are labelled items that fall outside (outer join) another reduced expression. 

I do get a warning which I really want to alert upon!

**Who is this feature for?**

For Grafana users who rely on Grafana-managed alerting, especially for agents that push metrics instead of polling them (e.g., telegraf + influxDB)

**Comment by santihernandezc** [comment]
Hi @kitos9112, could you provide an example so we can understand better what you're referring to?

**Comment by kitos9112** [comment]
> Hi @kitos9112, could you provide an example so we can understand better what you're referring to?

Of course! Let's suppose there is an influxdb datasource that holds dynamic timeseries data of production hosts that push data via telegraf. To keep it simple, Timeseries in influxdb are tagged with arbitrary host names. Additionally, another infinity datasource configured with in-line CSV data in the alert rule that replicates the reduced influxdb timeseries alongside their labels (Aka the influxdb host tag) 

By leveraging the Grafana dynamic/unified mutildimensional alerting model, we cannot tell easily enough when a host stops pushing `specific`timeseries data to influxDB. Not saying the host goes offline (we have liveness check for that). I'm alluding to one of the telegraf input plugins not collecting telemetry data correctly (e.g. missing series). 

In my mind I thought of just comparing the outter joint result of these two queries (A - > influxdb, B - > in line csv infinity) to alert upon. If these two don't match in size and content, then we should alert. 

I hope that makes sense, happy to clarify or hear other r better if approaches to tackle this use case.

**Comment by kitos9112** [comment]
@santihernandezc Does the above make any sense to triage and understand the initial requirements?

**Comment by rwwiv** [comment]
Yes, this helps clarify what you're looking for, thank you! We don't have existing plans to implement this, but do feel this is worth considering as a feature.

**Comment by kitos9112** [comment]
@rwwiv Touching base here to see if this can be prioritised or taken a look at?
